eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/listeners/ClusteringProgressListener.java

```diff
@@ -41,16 +60,59 @@ public ExitStatus afterStep(StepExecution stepExecution) {
+                clusteringCounts.getSubmittedVariantsUpdateOperationWritten());
+
+        try {
+            String assembly = stepExecution.getJobExecution().getJobParameters().getString("assemblyAccession");
+            String identifier = createIdentifier(assembly);
+            String url = stepExecution.getJobExecution().getJobParameters().getString("countStatsUrl") + URL_PATH_SAVE_COUNT;
+            saveClusteringCountMetricsInDB(url, numTotalItemsRead, clusteringCounts, identifier);
+        } catch (JSONException | RestClientException ex) {
+            logger.error("Error occurred while saving Counts to DB", ex);
+        }
+
         return status;
     }
+
+    private String createIdentifier(String assembly) throws JSONException {
+        JSONObject identifier = new JSONObject();
+        identifier.put("assembly", assembly);
+        return identifier.toString();
+    }
+
+    private void saveClusteringCountMetricsInDB(String url, long numTotalItemsRead, ClusteringCounts clusteringCounts, String identifier) {
+        List<Count> countList = new ArrayList<>();
+        countList.add(new Count(PROCESS, identifier, Metric.SUBMITTED_VARIANTS.getName(), numTotalItemsRead));
+        countList.add(new Count(PROCESS, identifier, Metric.CREATED_VARIANTS.getName(), clusteringCounts.getClusteredVariantsCreated()));
+        countList.add(new Count(PROCESS, identifier, Metric.UPDATED_VARIANTS.getName(), clusteringCounts.getClusteredVariantsUpdated()));
+        countList.add(new Count(PROCESS, identifier, Metric.MERGED_VARIANTS.getName(), clusteringCounts.getClusteredVariantsMergeOperationsWritten()));
+        countList.add(new Count(PROCESS, identifier, Metric.SUBMITTED_VARIANTS_CLUSTERED.getName(), clusteringCounts.getSubmittedVariantsClustered()));
+        countList.add(new Count(PROCESS, identifier, Metric.SUBMITTED_VARIANTS_UNCLUSTERED.getName(), clusteringCounts.getSubmittedVariantsKeptUnclustered()));
+        countList.add(new Count(PROCESS, identifier, Metric.SUBMITTED_VARIANTS_RS_MERGED.getName(), clusteringCounts.getSubmittedVariantsUpdatedRs()));
+        countList.add(new Count(PROCESS, identifier, Metric.SUBMITTED_VARIANTS_UPDATED_OPERATIONS.getName(), clusteringCounts.getSubmittedVariantsUpdateOperationWritten()));
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.setAccept(Collections.singletonList(MediaType.APPLICATION_JSON));
+
+        HttpEntity<Object> requestEntity = new HttpEntity<>(countList, headers);
+        ResponseEntity<List<Count>> response = restTemplate.exchange(url, HttpMethod.POST, requestEntity,
+                new ParameterizedTypeReference<List<Count>>() {
+                });
+
+        if (response.getStatusCode() == HttpStatus.OK) {
+            logger.info("Metric Count successfully saved In DB");
+        } else {
+            logger.warn("Could not save count In DB");
```

Comment:
Any reason why this shouldn't be an exception?

Reply:
Should be an error. Updated

Reply:
updated to throw a RestClientException